### PR TITLE
MF-209: Fixed links on height and weight not working

### PR DESCRIPTION
**Pull request by donaldkibet**
Presently the links under Results for Height and Weight bug out. This PR fixes this issue [MF-209](https://issues.openmrs.org/projects/MF/issues/MF-209)

MF-209

### CODE REVIEW

File: src/widgets/heightandweight/heightandweight.resource.ts [hunk 1]
```diff
@@ -15,8 +15,10 @@ function getDimensionsObservations(patientId: string) {
   return openmrsObservableFetch(
```
Comment (FlorianRappl) [inline src/widgets/heightandweight/heightandweight.resource.ts]: If we get the expected type of this request in there we don't need to be explicit with the typing below. Also we can get rid of `[]` and `any`, which give us no guarantees.

**Review by FlorianRappl — COMMENTED** [review]
LGTM so far just improve typing and then let's merge! :rocket:

File: src/widgets/heightandweight/heightandweight.resource.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ function getDimensionsObservations(patientId: string) {
     `${fhirConfig.baseUrl}/Observation?subject:Patient=${patientId}&code=${WEIGHT_CONCEPT},${HEIGHT_CONCEPT}`
   ).pipe(
     map(({ data }) => data["entry"]),
-    map((entries: []) => entries.map((entry: any) => entry.resource)),
+    map(entries => entries.map(entry => entry.resource)),
```
Comment (FlorianRappl) [inline src/widgets/heightandweight/heightandweight.resource.ts:19]: What I meant was that we should type the response to know exactly what `data` is. Then we know exactly whats on `data` and if we would type `entry.resourc` (missing 'e') the compiler would complain.

Right now its just empty and we need to do `data["entry"]` to go on. With `any`, which means no compiler assistance here.

Comment (donaldkibet) [reply src/widgets/heightandweight/heightandweight.resource.ts:19]: :+1:  thats makes it clear

**Comment by jonathandick** [comment]
@FlorianRappl could you take a look at @donaldkibet 's recent changes?

File: src/widgets/heightandweight/heightandweight.resource.ts [hunk 3]
```diff
@@ -64,3 +66,41 @@ function latestFirst(a, b) {
 function getDatesIssued(dimensionArray): string[] {
   return dimensionArray.map(dimension => dimension.issued);
 }
+
+type DimensionFetchResponse = {
+  entry: Array<Dimension>;
+  id: string;
+  resourceType: string;
+  total: Number;
```
Comment (FlorianRappl) [inline src/widgets/heightandweight/heightandweight.resource.ts]: The type is `number`. `Number` is the constructor.

File: src/widgets/heightandweight/heightandweight.resource.ts [hunk 4]
```diff
@@ -64,3 +66,41 @@ function latestFirst(a, b) {
...
+    resourceType: string;
+    status: string;
+    subject: {
+      display: string;
+      identifier: { id: string; system: string; use: string; value: string };
+      reference: string;
+      type: string;
+    };
+    valueQuantity: {
+      value: Number;
```
Comment (FlorianRappl) [inline src/widgets/heightandweight/heightandweight.resource.ts]: Same here.

File: src/widgets/heightandweight/heightandweight.resource.ts [hunk 5]
```diff
@@ -12,11 +12,13 @@ export function getDimensions(patientId: string) {
...
 
 function getDimensionsObservations(patientId: string) {
-  return openmrsObservableFetch(
+  return openmrsObservableFetch<DimensionFetchResponse>(
     `${fhirConfig.baseUrl}/Observation?subject:Patient=${patientId}&code=${WEIGHT_CONCEPT},${HEIGHT_CONCEPT}`
   ).pipe(
-    map(({ data }) => data["entry"]),
-    map((entries: []) => entries.map((entry: any) => entry.resource)),
+    map(({ data }) => data.entry),
+    map((entries: Array<Dimension>) =>
```
Comment (FlorianRappl) [inline src/widgets/heightandweight/heightandweight.resource.ts]: I think there should not be an explicit `Array<Dimension>` specified here, it should be correctly inferred by TS.

File: src/widgets/heightandweight/heightandweight.resource.ts [hunk 6]
```diff
@@ -12,11 +12,13 @@ export function getDimensions(patientId: string) {
...
+      entries.map((entry: Dimension) => entry.resource)
```
Comment (FlorianRappl) [inline src/widgets/heightandweight/heightandweight.resource.ts]: Also here. Please leave that up to TS. The types should just flow correctly.

**Review by FlorianRappl — COMMENTED** [review]
I think we are on a great way here.

Sorry for being so picky with the types. I do care a lot about correct typing, because it provides a valuable tool not only for future development, but also for maintenance and code understanding.

**Review by FlorianRappl — APPROVED** [review]
LGTM!

**Pull request merged**